modular_skyrat/modules/deforest_medical_items/code/chemicals/demoneye.dm

```diff
@@ -0,0 +1,152 @@
+		/datum/reagent/impurity/healing/medicine_failure = 10,
+		/datum/reagent/impurity = 5,
+	)
+	required_reagents = list(
+		/datum/reagent/medicine/ephedrine = 5,
+		/datum/reagent/blood = 15,
+		/datum/reagent/toxin/plasma = 5,
+	)
+	mob_react = FALSE
+	reaction_tags = REACTION_TAG_EASY | REACTION_TAG_DRUG | REACTION_TAG_ORGAN | REACTION_TAG_DAMAGING
+
+// Demoneye, a drug that makes you temporarily immune to fear and crit, in exchange for damaging all of your organs and making your veins explode
+/datum/reagent/drug/demoneye
+	name = "DemonEye"
+	description = "A performance enhancing drug originally developed on mars. \
+		A favorite among gangs and other outlaws on the planet, though overuse can cause terrible addiction and bodily damage."
+	reagent_state = LIQUID
+	color = "#af00be"
+	taste_description = "industrial shuttle fuel"
+	metabolization_rate = 0.65 * REAGENTS_METABOLISM
+	ph = 7
+	overdose_threshold = 15
+	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED
+	addiction_types = list(/datum/addiction/stimulants = 15)
+	/// How much time has the drug been in them?
+	var/constant_dose_time = 0
+
+
+/datum/reagent/drug/demoneye/on_mob_metabolize(mob/living/carbon/human/our_guy)
+	. = ..()
+
+	ADD_TRAIT(our_guy, TRAIT_UNNATURAL_RED_GLOWY_EYES, TRAIT_NARCOTICS)
+	// IM FUCKIN INVINCIBLE
+	ADD_TRAIT(our_guy, TRAIT_NOSOFTCRIT, TRAIT_NARCOTICS)
+	ADD_TRAIT(our_guy, TRAIT_NOHARDCRIT, TRAIT_NARCOTICS)
+	ADD_TRAIT(our_guy, TRAIT_FEARLESS, TRAIT_NARCOTICS)
+	ADD_TRAIT(our_guy, TRAIT_NUMBED, TRAIT_NARCOTICS)
+
+	our_guy.eye_color_left = BLOODCULT_EYE
+	our_guy.eye_color_right = BLOODCULT_EYE
+	our_guy.update_body()
+
+	our_guy.sound_environment_override = SOUND_ENVIRONMENT_PSYCHOTIC
+
+	if(!our_guy.hud_used)
+		return
+
+	var/atom/movable/plane_master_controller/game_plane_master_controller = our_guy.hud_used.plane_master_controllers[PLANE_MASTERS_GAME]
+
+	var/list/col_filter_red = list(0.7,0,0,0, 0,0,0,0, 0,0,0,0, 0,0,0,1)
+
+	game_plane_master_controller.add_filter("demoneye_filter", 10, color_matrix_filter(col_filter_red, FILTER_COLOR_RGB))
+
+	game_plane_master_controller.add_filter("demoneye_blur", 1, list("type" = "angular_blur", "size" = 4))
+
+	for(var/filter in game_plane_master_controller.get_filters("demoneye_blur"))
+		animate(filter, loop = -1, size = 2, time = 3 SECONDS, easing = ELASTIC_EASING|EASE_OUT, flags = ANIMATION_PARALLEL)
+		animate(size = 5, time = 3 SECONDS, easing = ELASTIC_EASING|EASE_IN)
+
+
+/datum/reagent/drug/demoneye/on_mob_end_metabolize(mob/living/carbon/human/our_guy)
+	. = ..()
+
+	REMOVE_TRAIT(our_guy, TRAIT_UNNATURAL_RED_GLOWY_EYES, TRAIT_NARCOTICS)
+	REMOVE_TRAIT(our_guy, TRAIT_NOSOFTCRIT, TRAIT_NARCOTICS)
+	REMOVE_TRAIT(our_guy, TRAIT_NOHARDCRIT, TRAIT_NARCOTICS)
+	REMOVE_TRAIT(our_guy, TRAIT_FEARLESS, TRAIT_NARCOTICS)
+	REMOVE_TRAIT(our_guy, TRAIT_NUMBED, TRAIT_NARCOTICS)
+
+	our_guy.eye_color_left = initial(our_guy.eye_color_left)
+	our_guy.eye_color_right = initial(our_guy.eye_color_right)
```

Comment:
This is the same thing cult red eyes does, and that seems to work fine on removal. I didn't notice anything wrong with eye colors while testing so I say it's probably fine?

Reply:
Really? That seems very bizarre to me... Alright, we'll see it on live, but keep a mental note about it, because I can see this becoming a bug in the future.